src/hooks/screen/useRootChannelListHook.ts

```diff
@@ -29,6 +40,76 @@ const useRootChannelListHook = () => {
     }
   };
 
+  const requestPermission = async () => {
+    const authStatus = await messaging().requestPermission();
+    const enabled =
+      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+      authStatus === messaging.AuthorizationStatus.PROVISIONAL;
+
+    if (enabled) {
+      const fcmToken = await messaging().getToken();
+      const payload = {
+        fcm_token: fcmToken
+      };
+      fcmTokenService(payload);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `requestPermission` function is requesting permission and then immediately sending the FCM token to the server. This could be a potential security issue if the user later decides to revoke the permission. It would be better to send the FCM token to the server only when it's needed, for example when the user interacts with a feature that requires push notifications.



<!-- This is an auto-generated comment by OSS CodeRabbit -->